traffic_control/clients/python/to_access/__init__.py

```diff
@@ -0,0 +1,455 @@
+	                         "and they need not start with 'http://' or 'https://'. HTTPS is the "
+	                         "assumed protocol unless the scheme _is_ provided and is 'http://'."))
+	parser.add_argument("--to-user",
+	                    type=str,
+	                    help="The username to use when connecting to Traffic Ops. Overrides '$TO_USER")
+	parser.add_argument("--to-password",
+	                    type=str,
+	                    help="The password to use when authenticating to Traffic Ops. Overrides '$TO_PASSWORD'")
+	parser.add_argument("-k", "--insecure", action="store_true", help="Do not verify SSL certificates")
+	parser.add_argument("-f", "--full",
+	                    action="store_true",
+	                    help=("Also output HTTP request/response lines and headers, and request payload. "
+	                         "This is equivalent to using '--request-headers', '--response-headers' "
+	                         "and '--request-payload' at the same time."))
+	parser.add_argument("--request-headers",
+	                    action="store_true",
+	                    help="Output request method line and headers")
+	parser.add_argument("--response-headers",
+	                    action="store_true",
+	                    help="Output response status line and headers")
+	parser.add_argument("--request-payload",
+	                    action="store_true",
+	                    help="Output request payload (will try to pretty-print if '--pretty' is given)")
+	parser.add_argument("-r", "--raw-path",
+	                    action="store_true",
+	                    help="Request exactly PATH; it won't be prefaced with '/api/{{api-version}}/")
+	parser.add_argument("-a", "--api-version",
+	                    type=float,
+	                    default=1.3,
+	                    help="Specify the API version to request against")
+	parser.add_argument("-p", "--pretty",
+	                    action="store_true",
+	                    help=("Pretty-print payloads as JSON. "
+	                         "Note that this will make Content-Type headers \"wrong\", in general"))
+	parser.add_argument("PATH", help="The path to the resource being requested - omit '/api/1.x'")
+	parser.add_argument("DATA",
+	                    help=("An optional data string to pass with the request. If this is a "
+	                         "filename, the contents of the file will be sent instead."),
+	                    nargs='?')
+
+
+	args = parser.parse_args()
+
+	try:
+		to_host = args.to_url if args.to_url else os.environ["TO_URL"]
+	except KeyError as e:
+		raise KeyError("Traffic Ops hostname not set! Set the TO_URL environment variable or use "\
+		               "'--to-url'.") from e
+
+	to_host = urlparse(to_host, scheme="https")
+	useSSL = to_host.scheme.lower() == "https"
+	to_port = to_host.port
+	if to_port is None:
+		if useSSL:
+			to_port = 443
+		else:
+			to_port = 80
+
+	port_pos = to_host.netloc.find(':')
```

Comment:
I thought I left a comment about this but I can't find it anymore. I think you can just use `to_host.host` here to get just the hostname without the `:port` suffix.

Reply:
I'm pretty sure you did, because I meant to respond to it with this:

I don't think so:
```
>>> from urllib.parse import urlparse
>>> urlparse("https://traffic-ops.company.com:443/")
ParseResult(scheme='https', netloc='traffic-ops.company.com:443', path='/', params='', query='', fragment='')
>>> urlparse("https://traffic-ops.company.com:443/").host
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'ParseResult' object has no attribute 'host'
>>> 
```

Reply:
my bad, try `to_host.hostname` instead

Reply:
dunno how I missed that